service/service_construction.go

```diff
@@ -786,8 +1092,35 @@ func parseErc20TransferData(data []byte) (*ethcommon.Address, *big.Int, error) {
 	return &address, amount, nil
 }
 
+func parseUnwrapData(data []byte) (*big.Int, *big.Int, error) {
+	if len(data) != genericUnwrapBytesLength {
+		return nil, nil, fmt.Errorf("incorrect length for data array")
+	}
+	methodID := getUnwrapMethodID()
+	if hexutil.Encode(data[:4]) != hexutil.Encode(methodID) {
+		return nil, nil, fmt.Errorf("incorrect methodID signature")
+	}
+
+	amount := new(big.Int).SetBytes(data[5:36])
+	chainID := new(big.Int).SetBytes(data[37:])
+
+	if chainID.Uint64() != 0 {
+		return nil, nil, fmt.Errorf("incorrect chainId value")
+	}
+
+	return amount, chainID, nil
+}
+
 func getTransferMethodID() []byte {
```

Comment:
I think a `getMethodID` function that takes in a fn signature would be good here

Reply:
#101